brunton_lab_to_nwb/brunton_lab_to_nwb.py

```diff
@@ -302,9 +302,9 @@ def get_data(label):
 
 def convert_dir(in_dir, events_path, r2_path, coarse_events_path, reach_features_path,
                 n_jobs=1, overwrite: bool = False):
-    all_files = Path(in_dir).iterdir()
-    all_data_files = [x.stem for x in all_files if ".h5" in x.suffix]
-    nwb_files = [x.stem for x in all_files if ".nwb" in x.suffix]
+    in_dir = Path(in_dir)
+    all_data_files = [x.stem for x in in_dir.iterdir() if ".h5" in x.suffix]
+    nwb_files = [x.stem for x in in_dir.iterdir() if ".nwb" in x.suffix]
```

Comment:
`.iterdir()` returns an iterator object, and these act like pointers. If you have this set as a variable that is then actually iterated over once (here, by list compression for `all_data_files `), then all the nested internal calls of `next()` on that iterator will update the original object. Hence, attempting to use that same iterator again for the same type of iteration, but for a different compression, fails since that iterator object has already concluded iterating.